### ISSUE TITLE
Problem: xmr-stak-cpu fails to build on non-glibc linux/bsd systems

### ISSUE THREAD

**Issue by thunderlark** [fireice-uk/xmr-stak-cpu#158]
Solution: change `#if defined(__APPLE__)` to `#if defined(__APPLE__) || defined(__FreeBSD__) || !defined(_GNU_SOURCE) || !defined(__GLIBC__)` in function `sock_strerror()` in file `socks.h`.

**Comment by psychocrypt** [comment]
IMO this solution has side effects to other compiler, I thing it should be 
```
#if ( defined(__APPLE__) || defined(__FreeBSD__) ) && ( !defined(_GNU_SOURCE) || !defined(__GLIBC__) )
```
Than only APLLE and FREEBSD is effected by the change

**Comment by thunderlark** [comment]
What's the problem with other compilers?

I don't think your version would be a comprehensive solution. For example, in Linux distributions without glibc (e.g. Alpine), this code would fail to prevent a nonstandard function from being used.

**Comment by psychocrypt** [comment]
In your suggestest fix a compiler which has not defined __GLIBC__ will enter the code path for apple/bsd, I will avoid such side effects and only enter the bsd path if the system has defined bsd or apple.

**Comment by thunderlark** [comment]
but the compiler doesn't define it, the header files for glibc do. and ALL of them will define macro `__GLIBC__`, trust me.

in any case, entering the bsd path is not a big deal. it still works fine and complies with standards. in fact i don't know why that isn't the default way of doing it.

**Comment by psychocrypt** [comment]
I also checked it again and it looks like you are right, #159 will introduce the changes.
I need to verify it with clang first before I can metge it.

**Comment by GremL1N** [comment]
i have next trouble on BSD
minethd.cpp:55:2: error: unknown type name
      'cpu_set_t'; did you mean 'cpusetid_t'?

**Comment by psychocrypt** [comment]
@GremL1N  It could be that the type naming is different on BSD . In the development branch `dev` we have already a fix for FreeBSD [link](https://github.com/fireice-uk/xmr-stak-cpu/blob/dev/minethd.cpp#L59), please use the dev branch up to the next release.

**Comment by psychocrypt** [comment]
initial issue should be fixed with #159
